tests/main.cpp

```diff
@@ -137,4 +137,497 @@ TEST_CASE("Test getMultipleAccounts") {
   REQUIRE_NE(accountInfoMap.size(), accounts.size());
   it = accountInfoMap.find("9aZg1jhgRzGRmYWLbTrorCFE7BQbaz2dE5nYKmqeLGCW");
   CHECK_EQ(it, accountInfoMap.end());
+}
+TEST_CASE("Empty MangoAccount") {
+  std::string resources_dir = FIXTURES_DIR;
+  auto path = resources_dir + "/mango_v3/empty";
+  auto mangoGroup =
+      solana::rpc::fromFile<mango_v3::MangoGroup>(path + "/group.json");
+  auto accountInfo =
+      solana::rpc::fromFile<mango_v3::MangoAccountInfo>(path + "/account.json");
+  auto mangoAccount = mango_v3::MangoAccount(accountInfo);
+  auto mangoCache =
+      solana::rpc::fromFile<mango_v3::MangoCache>(path + "/cache.json");
+  auto initHealth = mangoAccount.getHealth(mangoGroup, mangoCache,
+                                           mango_v3::HealthType::Init);
+  CHECK_EQ(initHealth.to_double(), 0);
+  auto maintHealth = mangoAccount.getHealth(mangoGroup, mangoCache,
+                                            mango_v3::HealthType::Maint);
+  CHECK_EQ(maintHealth.to_double(), 0);
+  auto initRatio = mangoAccount.getHealthRatio(mangoGroup, mangoCache,
+                                               mango_v3::HealthType::Init);
+  CHECK_EQ(initRatio.to_double(), 100);
+  auto maintRatio = mangoAccount.getHealthRatio(mangoGroup, mangoCache,
+                                                mango_v3::HealthType::Maint);
+  CHECK_EQ(maintRatio.to_double(), 100);
+  auto value = mangoAccount.computeValue(mangoGroup, mangoCache);
+  CHECK_EQ(value.to_double(), 0);
+  auto leverage = mangoAccount.getLeverage(mangoGroup, mangoCache);
+  CHECK_EQ(leverage.to_double(), 0);
+  CHECK_FALSE(mangoAccount.isLiquidatable(mangoGroup, mangoCache));
+}
+
+TEST_CASE("1deposit") {
+  std::string resources_dir = FIXTURES_DIR;
+  auto path = resources_dir + "/mango_v3/1deposit";
+  auto mangoGroup =
+      solana::rpc::fromFile<mango_v3::MangoGroup>(path + "/group.json");
+  auto accountInfo =
+      solana::rpc::fromFile<mango_v3::MangoAccountInfo>(path + "/account.json");
+  auto mangoAccount = mango_v3::MangoAccount(accountInfo);
+  auto mangoCache =
+      solana::rpc::fromFile<mango_v3::MangoCache>(path + "/cache.json");
+  auto initHealth = mangoAccount.getHealth(mangoGroup, mangoCache,
+                                           mango_v3::HealthType::Init);
+  CHECK_EQ(initHealth.to_double(), 37904260000.059059);
```

Comment:
how did you generate these results?

Reply:
Hey @mschneider these values are copied from the TS SDK. Specifically this PR: https://github.com/blockworks-foundation/mango-client-v3/pull/50